cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -1590,7 +1705,13 @@ const FunctionDoc strftime_doc{
      "does not exist on this system."),
     {"timestamps"},
     "StrftimeOptions"};
-
+const FunctionDoc strptime_doc(
+    "Parse timestamps",
+    ("For each string in `strings`, parse it as a timestamp.\n"
+     "The timestamp unit and the expected string pattern must be given\n"
+     "in StrptimeOptions.  Null inputs emit null.  If a non-null string\n"
+     "fails parsing, an error is returned by default."),
```

Comment:
Ditto here - what is the behavior if we don't raise errors and we get an invalid timestamp?

Reply:
Done. Please check.